### PR TITLE
feat: Add TextArea component

### PR DESCRIPTION
**Pull request by robiharid**
image

### CODE REVIEW

File: src/components/TextArea/TextArea.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
+<template>
+  <component
+    :is="formElement ? CFormElement : CFragment"
+    v-bind="{ id, label, size, color, labelFormatter, message }"
```
Comment (mattbarton44) [inline src/components/TextArea/TextArea.vue:4]: Need to pass stacked in too for FormElement
```suggestion
    v-bind="{ id, label, size, color, labelFormatter, message, stacked }"
```

File: src/components/TextArea/TextArea.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+<script setup>
+import { computed, ref } from 'vue';
+import {
+  colorProp,
+  formElementProps,
+  useSizeProp,
+} from '../../composables/props.js';
+import { useInputColorClassValue } from '../../composables/styles.js';
+import { useSizeValue } from '../../composables/forms';
```
Comment (mattbarton44) [inline src/components/TextArea/TextArea.vue:49]: forms contain UseStackedValue and useFormElementValue which can both be used the same way as useSizeValue to be bound to formElement

File: src/components/TextArea/TextArea.vue [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  transparent: { type: Boolean, default: false },
+
+  onEnter: { type: Function, default: null },
+  onBlur: { type: Function, default: null },
+  rows: { type: Number, default: 10 },
+});
+
+const emit = defineEmits(['update:modelValue', 'enter', 'blur']);
+
+const formElement = props.isFormElement || !!props.label;
```
Comment (mattbarton44) [inline src/components/TextArea/TextArea.vue:73]:
```suggestion
const formElement = useFormElementValue(props.label);
```

**Review by mattbarton44 — CHANGES_REQUESTED** [review]

**Comment by robiharid** [comment]
Brill, thanks @mattbarton44 for clearing up the use case for the composable useFormElementValue. 
All suggestions are reflected for you now. :)

**Review by danielstorey — APPROVED** [review]

File: src/components/TextArea/TextArea.vue [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+</template>
+
+<script setup>
+import { computed, ref } from 'vue';
+import {
+  colorProp,
+  formElementProps,
+  useSizeProp,
+} from '../../composables/props.js';
+import { useFormElementValue } from '../../composables/forms';
```
Comment (mattbarton44) [inline src/components/TextArea/TextArea.vue:48]:
```suggestion
import { useSizeValue, useStackedValue, useFormElementValue } from '../../composables/forms';
```

File: src/components/TextArea/TextArea.vue [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  transparent: { type: Boolean, default: false },
+
+  onEnter: { type: Function, default: null },
+  onBlur: { type: Function, default: null },
+  rows: { type: Number, default: 10 },
+});
+
+const emit = defineEmits(['update:modelValue', 'enter', 'blur']);
+
+const formElement = useFormElementValue(props.label);
```
Comment (mattbarton44) [inline src/components/TextArea/TextArea.vue:74]:
```suggestion
const stacked = useStackedValue(props.stacked);
const formElement = useFormElementValue(props.label);
```

File: src/components/TextArea/TextArea.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  get() {
+    return props.modelValue;
+  },
+  set(value) {
+    isDirty.value = true;
+    emit('update:modelValue', value);
+  },
+});
+
+const size = useFormElementValue(props.size);
```
Comment (mattbarton44) [inline src/components/TextArea/TextArea.vue:87]:
```suggestion
const size = useSizeValue(props.size);
```

**Review by mattbarton44 — CHANGES_REQUESTED** [review]

**Review by mattbarton44 — APPROVED** [review]

**Pull request merged**